src/wallet/wallet_rpc_server.cpp

```diff
@@ -385,7 +385,7 @@ namespace tools
       entry.destinations.push_back(wallet_rpc::transfer_destination());
       wallet_rpc::transfer_destination &td = entry.destinations.back();
       td.amount = d.amount;
-      td.address = get_account_address_as_str(m_wallet->nettype(), d.is_subaddress, d.addr);
+      td.address = d.original.empty() ? get_account_address_as_str(m_wallet->nettype(), d.is_subaddress, d.addr) : d.original;
```

Comment:
Why do we need this change? 

d.original could be a url (openalias??) from a quick look at `wallet_rpc_server::validate_transfer` https://github.com/monero-project/monero/blob/914b1061d49c98b7966a7c6f75dddf2bed159f52/src/wallet/wallet_rpc_server.cpp#L766 & https://github.com/monero-project/monero/blob/914b1061d49c98b7966a7c6f75dddf2bed159f52/src/wallet/wallet_rpc_server.cpp#L787

Reply:
We don't really need it. It's nice to the user though. Many users are confused when they get back a different address than what they used, which is understandable.

Reply:
Are you implying you think it's better to return canonical rather than original ?

Reply:
Yes, but I also see how some may prefer the supplied. Thus not sure why this change added in for a patch for a test when it will affect more than just a test. Anyone relying on existing behavior must factor this change in.

Reply:
Many people were surprised when they got a different address than what they used. If you don't know how integrated addresses work for example, you may not even know the address you sent to can be seen under another guise. I added a test for it because I guess I like adding tests for stuff when I can these days, though I don't always do it :)
Fair point about change of expectations, though if someone is relying on the API returning a different address than the one that was input, then I'd be pretty surprised.

Reply:
Yeah fair enough. Tests are of course good :)